packages/manager/apps/web/client/app/hosting/advices/hosting-pro/advice-hosting-pro.controller.js

```diff
@@ -0,0 +1,148 @@
+import compact from 'lodash/compact';
+import endsWith from 'lodash/endsWith';
+import filter from 'lodash/filter';
+import get from 'lodash/get';
+
+export default class {
+  /* @ngInject */
+  constructor(
+    $http,
+    $q,
+    $state,
+    $translate,
+    Hosting,
+    OvhApiMe,
+    RedirectionService,
+    CORE_URLS,
+  ) {
+    this.$http = $http;
+    this.$q = $q;
+    this.$state = $state;
+    this.$translate = $translate;
+    this.Hosting = Hosting;
+    this.OvhApiMe = OvhApiMe;
+    this.RedirectionService = RedirectionService;
+
+    this.CORE_URLS = CORE_URLS;
+  }
+
+  $onInit() {
+    this.advices = this.loadAdvices();
+  }
+
+  loadAdvices() {
+    return this.OvhApiMe.v6()
+      .get()
+      .$promise.then((user) =>
+        this.$q
+          .all([
+            this.getEmailPro(user),
+            this.getWebPerformance(),
+            this.getECommerceDomains(user),
+            this.getSms(),
+            this.getDnsAnycast(),
+            this.getCdn(),
+          ])
+          .then((advices) => compact(advices)),
+      );
+  }
+
+  getEmailPro(user) {
+    return this.$http.get('/email/pro').then(
+      (response) =>
+        !response.data.length
+          ? {
+              localizedName: this.$translate.instant(
+                'hosting_pro_advices_email_pro',
+              ),
+              href: get(
+                this.CORE_URLS,
+                `website_order.email_pro.${user.ovhSubsidiary}`,
+              ),
+              tag: 'cross_sell::hosting::clients_with_prowebhosting::email_pro',
+              external: true,
+            }
+          : null,
+      () => null,
+    );
+  }
+
+  getWebPerformance() {
+    return {
+      localizedName: this.$translate.instant(
+        'hosting_pro_advices_web_performance',
+      ),
+      href: this.$state.href('app.hosting.upgrade', {
```

Comment:
all the href can be input to the component, this way you can remove the dependency on $state

Reply:
This is no longer needed, since the urls have been moved to the 2APIs